### DISCUSSION TITLE
Why is changing the font size so difficult?

### DISCUSSION THREAD

**Discussion by karypid** [eclipse-platform/eclipse.platform.ui#447] [General]
Hello,

Due to my laptop's resolution, I need to increase the font size. While I can easily do that for some parts of Eclipse, it seems impossible for the project explorer view.

I have searched for resources and found:

- https://riptutorial.com/eclipse/example/26874/how-to-configure-the-font-size-of-views-in-eclipse-on-linux
- https://stackoverflow.com/questions/3124629/how-can-i-configure-the-font-size-for-the-tree-item-in-the-package-explorer-in-e
- https://stackoverflow.com/questions/4466247/eclipse-changing-font-size-in-project-package-explorer
- https://stackoverflow.com/questions/45652472/how-to-increase-font-of-menu-bar-in-eclipse

My first observation was that there seems to be a platform-specific aspect. I am on Windows so need something that works there.

The second thing is: why is this so hard? I have tried editing the CSS file "e4-dark_win.css" since I use the dark theme and managed to change some parts with:

```
MPart Tree {
    font-size: 12;
}
```

 This made the problem view and parts of the package explorer bigger, but not all of the content gets the font applied. For example, the "JRE System Library" and "Maven dependencies" entries use the bigger font I specified, but the regular folders/files are still shown with the original small font. I think that everything that has an annotation on its icon from source control (git) ignore my font setting...

What can I do to fix that?

I'm a bit surprised that this is so difficult...

**Comment by BeckerWdf** [comment]
Besides eclipse preferences / SWT styling eclipse platform also reads font settings from the operating system. Changing font sizes in Windows preferences might help.
